subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2261,18 +2571,62 @@ static int do_read_op(struct lwm2m_engine_obj *obj,
 	return ret;
 }
 
-static int do_discover_op(struct lwm2m_engine_context *context)
+static int print_attr(struct net_pkt *pkt, char *buf, u16_t buflen,
+		      sys_slist_t *attr_list)
+{
+	struct lwm2m_attr *attr;
+	int used, base = 100000;
+	u8_t digit;
+	s32_t dec;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(attr_list, attr, node) {
+		/* assuming integer will have float_val.val2 set as 0 */
+		used = snprintk(buf, buflen, ";%s=%d%s",
+				LWM2M_ATTR_STR[attr->type],
+				attr->float_val.val1,
+				attr->float_val.val2 > 0 ? "." : "");
+
+		dec = attr->float_val.val2;
+		while (dec && used < buflen && base > 0) {
+			digit = dec / base;
+			buf[used++] = '0' + digit;
+			dec -= digit * base;
+			base /= 10;
+		}
+
+		if (!net_pkt_append_all(pkt, used, buf, BUF_ALLOC_TIMEOUT)) {
+			return -ENOMEM;
+		}
+	}
+
+	return 0;
+}
+
+static int do_discover_op(struct lwm2m_engine_context *context, bool well_known)
 {
 	static char disc_buf[24];
```

Comment:
Unrelated to the patch, but is there a way that do_discover_op() can be called from two different threads?